### ISSUE TITLE
gpg: fatal: can't create directory `//.gnupg': Permission denied

### ISSUE THREAD

**Issue by trevmurray** [wal-e/wal-e#163]
I've injected the WAL-E module into a postgres docker container. I can get the streaming archiving to S3 to work when I don't incorporate encryption. However, I need to encrypt the files prior to shipping to S3. I generated a gpg key, created a WALE_GPG_KEY_ID with the key id enclosed. While attached to the container, I executed the following command: su - postgres bash -c "envdir /etc/wal-e.d/env wal-e backup-push $PGDATA"

I receive the following error:
wal-e: Could not set up syslog, continuing anyway.  Reason: ENOENT
wal_e.operator.backup INFO     MSG: start upload postgres version metadata
        DETAIL: Uploading to s3://sofi-postgresql-backups/basebackups_005/base_000000010000000000000002_00000040/extended_version.txt.
        STRUCTURED: time=2015-01-28T19:34:09.320846-00 pid=31
wal_e.operator.backup INFO     MSG: postgres version metadata upload complete
        STRUCTURED: time=2015-01-28T19:34:09.395145-00 pid=31
wal_e.worker.upload INFO     MSG: beginning volume compression
        DETAIL: Building volume 0.
        STRUCTURED: time=2015-01-28T19:34:09.523195-00 pid=31
gpg: fatal: can't create directory `//.gnupg': Permission denied
secmem usage: 0/0 bytes in 0/0 blocks of pool 0/32768

Here is the log from the "docker logs <container>" command:

wal-e: Could not set up syslog, continuing anyway.  Reason: ENOENT
wal_e.worker.upload INFO     MSG: begin archiving a file
        DETAIL: Uploading "pg_xlog/000000010000000000000001" to "s3://sofi-postgresql-backups/wal_005/000000010000000000000001.lzo".
        STRUCTURED: time=2015-01-28T19:54:13.844163-00 pid=173 action=push-wal key=s3://sofi-postgresql-backups/wal_005/000000010000000000000001.lzo prefix= seg=000000010000000000000001 state=begin
wal_e.worker.upload INFO     MSG: begin archiving a file
        DETAIL: Uploading "pg_xlog/000000010000000000000002" to "s3://sofi-postgresql-backups/wal_005/000000010000000000000002.lzo".
        STRUCTURED: time=2015-01-28T19:54:13.864128-00 pid=173 action=push-wal key=s3://sofi-postgresql-backups/wal_005/000000010000000000000002.lzo prefix= seg=000000010000000000000002 state=begin
gpg: fatal: can't create directory `/root/.gnupg': Permission denied
secmem usage: 0/0 bytes in 0/0 blocks of pool 0/32768

Seems to be a permissions issue with the /root/.gnupg directory during encryption. I've altered the permissions to no avail. Any direction would be greatly appreciated.

**Comment by trevmurray** [comment]
Here are other errors of note within the log: 

wal_e.main   CRITICAL MSG: pipeline process did not exit gracefully
        DETAIL: "gpg -e -z 0 -r psql@sofi.com" had terminated with the exit status 2.
        STRUCTURED: time=2015-01-28T19:54:14.057127-00 pid=173
LOG:  archive command failed with exit code 1
DETAIL:  The failed archive command was: envdir /etc/wal-e.d/env /usr/local/bin/wal-e wal-push pg_xlog/000000010000000000000001

**Comment by fdr** [comment]
Try running GPG on your own, e.g. from the command line.  Something is missing, but regretfully I don't have a firm idea what that may be.  The `//.gnupg` string is suspicious: it seems like that's a null-expansion of some kind.

**Comment by trevmurray** [comment]
Resolved my issue. The full-backup was running as the postgres user and using the /home/postgres/.gnupg dir. However the wal-push command (which was added to the archive_command setting in postgresql.conf) was being ran as the postgres user the the command to run as the postgres user was being executed by root using the /root/.gnupg directory which access was denied. Within the container I changed roots env home to /home/postgres/ and thereafter the .gnupg directory was properly referenced. Hope that makes sense.
